python/ray/data/_internal/planner/map_batches.py

```diff
@@ -110,7 +64,7 @@ def process_next_batch(batch: DataBatch) -> Iterator[Block]:
             block_iter = iter(blocks)
             first_block = next(block_iter)
             blocks = itertools.chain([first_block], block_iter)
-            empty_block = BlockAccessor.for_block(first_block).builder().build()
+            empty_block = BlockAccessor.for_block(first_block).slice(0, 0, copy=True)
```

Comment:
cc @amogkam for the change, FYI.

Reply:
@amogkam The previous way only preserves the format, but will lose the schema.